src/Internal/Client/WorkflowStarter.php

```diff
@@ -335,7 +340,8 @@ private function configureExecutionRequest(
             ->setWorkflowTaskTimeout(DateInterval::toDuration($options->workflowTaskTimeout))
             ->setMemo($options->toMemo($this->converter))
             ->setSearchAttributes($options->toSearchAttributes($this->converter))
-            ->setHeader($header->toHeader());
+            ->setHeader($header->toHeader())
+            ->setUserMetadata($metadata);
```

Comment:
You want to avoid setting it unless it actually contains something. Same goes for defaulting to empty strings. Proto will distinguish between unset metadata, and set metadata with empty fields. We don't want to actually set it unless something inside it has explicitly been set.

Reply:
What about Run/Execution/Task Timeouts? Is it better to skip zero Durations or set to `{seconds: 0, nanos:0}`?
I have already encountered instances where Temporal Cloud returns empty intervals as `null`, unlike the dev server. Should the SDK do the same now?

Reply:
Generally speaking we should avoid setting any field unless the user explicitly configured it